stdlib/http/src/main/java/org/ballerinalang/net/http/compiler/ResourceSignatureValidator.java

```diff
@@ -96,5 +96,11 @@ public static void validateAnnotation(FunctionNode resourceNode, DiagnosticLog d
 
     private ResourceSignatureValidator() {
     }
+
+    static void validateReturn(boolean resourceReturnsErrorOrNil, DiagnosticLog dlog, DiagnosticPos pos) {
+        if (!resourceReturnsErrorOrNil) {
+            dlog.logDiagnostic(Diagnostic.Kind.ERROR, pos, "invalid return type: expected error|()");
```

Comment:
Shall we change `error|()` to `error?` to be consistent with other compile errors?

Reply:
e.g., `error: .::test.bal:8:12: incompatible types: expected 'error?', found 'int'`